### PR TITLE
feat: refactor future handling

### PR DESCRIPTION
**Pull request by sethyates**
## Description of this change

Refactor future handling to fix some race conditions
Add cache handlers

## Why is this change being made?
- [ ] Bug fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How was this tested? How can the reviewer verify your testing?

Unit tests

## Related issues

N/A

## Checklist

- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.
- [X] I have evaluated the security impact of this change, and [OWASP Secure Coding Practices](https://owasp.org/www-project-secure-coding-practices-quick-reference-guide/migrated_content#) have been observed.
- [X] I have informed stakeholders of my changes.

### CODE REVIEW

File: src/ketch.ts [hunk 1]
```diff
@@ -1101,14 +1112,15 @@ export class Ketch extends EventEmitter {
 
     try {
       const region = await this.loadRegionInfo()
-      const jurisdiction = ps.scopes && ps.scopes[region] ? ps.scopes[region] : ps.defaultScopeCode || ''
+      log.info('ps', ps, region)
```
Comment (andolojm) [inline src/ketch.ts:1115]: Is leaving this logger in intentional?

File: src/ketch.ts [hunk 2]
```diff
@@ -1397,4 +1409,66 @@ export class Ketch extends EventEmitter {
     }
     return super.emit(event, ...args)
   }
+
+  addListener(eventName: string | symbol, listener: (...args: any[]) => void): this {
+    return this.on(eventName, listener)
+  }
+
+  on(eventName: string | symbol, listener: (...args: any[]) => void): this {
+    const f = this.mapEvent(eventName)
```
Comment (andolojm) [inline src/ketch.ts:1418]: Nitpick: these variables named `f` could be made clearer to help give me context on what the function is trying to accomplish

File: webpack.config.js [hunk 3]
```diff
@@ -14,17 +14,18 @@ module.exports = {
   resolve: {
     extensions: ['.ts', '.json', '.js'],
     fallback: {
-      'crypto': require.resolve('./src/polyfill/crypto.ts'),
-      'util': require.resolve('util/'),
-    }
+      crypto: require.resolve('./src/polyfill/crypto.ts'),
+      // util: require.resolve('util/'),
+      // 'process/browser': require.resolve('process/browser'),
```
Comment (andolojm) [inline webpack.config.js:19]: Are these paths unused?

**Review by andolojm — APPROVED** [review]
Approved w/ small nitpicks

**Review by keepforever — APPROVED** [review]

File: webpack.config.js (hunk 3 above)
Comment (sethyates) [reply webpack.config.js:19]: Seems like it. I built and ran the tag in the browser without these polyfills.

**Pull request merged**